app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/polygon/DrawAreaTaskViewModel.kt

```diff
@@ -210,18 +217,33 @@ internal constructor(
         if (vertices.isEmpty()) {
           null
         } else {
-          Feature(
-            id = uuidGenerator.generateUuid(),
-            type = FeatureType.USER_POLYGON.ordinal,
-            geometry = LineString(vertices),
-            style = Feature.Style(strokeColor, Feature.VertexStyle.CIRCLE),
-            clusterable = false,
-            selected = true,
-          )
+          buildPolygonFeature()
         }
       )
     }
 
+  private suspend fun buildPolygonFeature() =
+    Feature(
+      id = uuidGenerator.generateUuid(),
+      type = FeatureType.USER_POLYGON.ordinal,
+      geometry = LineString(vertices),
+      style = Feature.Style(strokeColor, Feature.VertexStyle.CIRCLE),
+      clusterable = false,
+      selected = true,
+      tooltipText = getDistanceTooltipText(),
+    )
+
+  /**
+   * Returns the distance tooltip if there is more than one vertex, if the last two vertices do not
+   * overlap, and if the polygon has not yet been marked as complete by the user.
```

Comment:
It is not recommended to write the low-level implementation details in the method doc. Instead it should reflect what the function is doing. In this case, it can look something like:

```
Returns the distance in meters between the last two vertices for displaying in the tooltip.
```